src/main/java/edu/harvard/iq/dataverse/authorization/providers/oauth2/oidc/OpenIDAuthentication.java

```diff
@@ -0,0 +1,46 @@
+package edu.harvard.iq.dataverse.authorization.providers.oauth2.oidc;
+
+import java.io.IOException;
+
+import edu.harvard.iq.dataverse.util.SystemConfig;
+import fish.payara.security.annotations.LogoutDefinition;
+import fish.payara.security.annotations.OpenIdAuthenticationDefinition;
+import fish.payara.security.openid.api.OpenIdConstant;
+import jakarta.annotation.security.DeclareRoles;
+import jakarta.ejb.EJB;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.annotation.HttpConstraint;
+import jakarta.servlet.annotation.ServletSecurity;
+import jakarta.servlet.annotation.WebServlet;
+import jakarta.servlet.http.HttpServlet;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+
+/**
+ * OIDC login implementation
+ */
+@WebServlet("/oidc/login")
+@OpenIdAuthenticationDefinition(
+    providerURI = "#{openIdConfigBean.providerURI}",
+    clientId = "#{openIdConfigBean.clientId}",
+    clientSecret = "#{openIdConfigBean.clientSecret}",
+    redirectURI = "#{openIdConfigBean.redirectURI}",
+    logout = @LogoutDefinition(redirectURI = "#{openIdConfigBean.logoutURI}"),
+    scope = {OpenIdConstant.OPENID_SCOPE, OpenIdConstant.EMAIL_SCOPE, OpenIdConstant.PROFILE_SCOPE},
+    tokenAutoRefresh = true,
+    useSession = true // If enabled state & nonce value stored in session otherwise in cookies.
+)
+@DeclareRoles("all")
+@ServletSecurity(@HttpConstraint(rolesAllowed = "all"))
+public class OpenIDAuthentication extends HttpServlet {
```

Comment:
It is because of the hack with the `rolesAllowed = "all"`, bearer token identity store hack and the "SecurityContext" hack. To discuss in the tech hours I think.